### PR TITLE
tools/trap: Update TRAP tool as per latest assert logs

### PR DESCRIPTION
**Pull request by thapav**
This patch updates the parser as per latest assert logs in arch files.

vidisha@vidisha:~/rt/tools/trap(master)$ python3 ramdumpParser.py -t log

*************************************************************
dump_file                   : None
log_file                    : log
Number of binary            : 1 [kernel]
"kernel" elf_file           : ../../build/output/bin/tinyara.axf
*************************************************************

DEBUG_MM_HEAPINFO is not enable. Enable DEBUG_MM_HEAPINFO to see heap usage -----------------------------------------------------------------------------------------
1. Crash Binary             : kernel

2. Crash type               : code assertion by code ASSERT or PANIC

3. Crash point
	-  print_assert_detail: Assertion failed at file:hello_main.c line: 71 task: appmain

4. Code asserted in:

	- Code asserted in normal thread.

5. Call stack of last run thread

Stack_address	 Symbol_address	 Symbol location  Symbol_name		File_name
 User stack
0x1006f8	 0xe003d9c	 kernel  	  vfprintf	/root/tizenrt/os/include/stdio.h:479 0x100700	 0xe003dd2	 kernel  	  stdoutstream_flush	/root/tizenrt/lib/libc/stdio/lib_stdoutstream.c:100 0x1006e20c	 0xe08a9ed	 kernel  	  __FUNCTION__.5309 0x1006e23c	 0xe006bcc	 kernel  	  hello_main	/root/tizenrt/apps/examples/hello/hello_main.c:67 0x1006e240	 0xe001afc	 kernel  	  task_start	/root/tizenrt/os/kernel/task/task_start.c:133

h. Heap Region information:

	!!!! NO HEAP CORRUPTION DETECTED !!!!

Checking kernel heap for corruption
Checking corruption (0x1005b260 - 0x10076ff8) : Heap NOT corrupted Checking corruption (02002130 - 0x023bfff8) : Heap NOT corrupted No Kernel heap corruption detected.

x. Miscellaneous information:

List of all tasks in the system:

                          NAME |   PID |  PRI | TOTAL STACK | STACK ALLOC ADDR | TCB ADDR
 -----------------------------------------------------------------------------------------------
                       Idle Task |     0 |    0 |    1024 |       0x10004bfc | 0x1002bd80
                          hpwork |     1 |  224 |  2028 |       0x1005dcf0 | 0x1005d690
                          lpwork |     2 |   50 |    2028 |       1005eb60 | 0x1005e500
                     LWIPCP/IP |     3 |  105 |    4068 |       0x100660f0 | 0x10065a90
            netmgr_event_handler |     4  100 |    2012 |       0x10067760 | 0x10067100
                           tash |     6 |  125 |    4076 |       0x10069470 | 0x10068e10
               ifi msg handler |     7 |  100 |    4060 |       0x1006af90 | 1006a9d0
                 ble msg handler |     8 |  100 |    4068 |       0x1006c600 | 0x1006bfa0
                         appmain |     9 |  100 |    28 |       0x1006da60 | 0x10068900
----------------------------------------------------------------------------------------------------

### CODE REVIEW

File: tools/trap/ramdumpParser.py [hunk 1]
```diff
@@ -600,7 +600,7 @@ def find_crash_point(log_file, elf):
                     continue
 
             # Get the assert location PC value
-            if 'up_assert: Assertion failed at file:' in line:
+            if 'print_assert_detail: Assertion failed at file:' in line:
```
Comment (sunghan-chang) [inline tools/trap/ramdumpParser.py:603]: How about checking both? or ignoring prefix?
Important (necessary) sentence is after `:`.
If we change the function name, we should modify this again. But it's meanless message in the TRAP.

Comment (sunghan-chang) [reply tools/trap/ramdumpParser.py:603]: @thapav Could you let me know ETA for this?

Comment (thapav) [reply tools/trap/ramdumpParser.py:603]: Yes your suggestions is a good one. I have incorporated the changes.

**Review by sunghan-chang — COMMENTED** [review]
There are many lines which chec function name like `up_assert :` or `task_dumptask:` or etc.
There is no issue with them?

**Comment by thapav** [comment]
> There are many lines which chec function name like `up_assert :` or `task_dumptask:` or etc. There is no issue with them?

Yes that could be an issue as well, I will update that ...

**Comment by sunghan-chang** [comment]
@thapav Please let me know ETA for this.

**Comment by thapav** [comment]
> @thapav Please let me know ETA for this.

Done

**Review by sunghan-chang — COMMENTED** [review]
How about line 1076 and 1078?
```
                    if 'task_show_alivetask_list:' in line:
                        start_idx += len('task_show_alivetask_list:')
                    if 'task_taskdump:' in line:
                        start_idx += len('task_taskdump:')
```

**Comment by thapav** [comment]
> How about line 1076 and 1078?
> 
> ```
>                     if 'task_show_alivetask_list:' in line:
>                         start_idx += len('task_show_alivetask_list:')
>                     if 'task_taskdump:' in line:
>                         start_idx += len('task_taskdump:')
> ```

Updated, Kindly check

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**